tez-plugins/tez-protobuf-history-plugin/src/main/java/org/apache/tez/dag/history/logging/proto/ProtoMessageWritable.java

```diff
@@ -96,6 +96,9 @@ public void readFields(DataInput in) throws IOException {
       cin = CodedInputStream.newInstance(din);
       cin.setSizeLimit(Integer.MAX_VALUE);
     }
+    if (din.in != in) {
+      cin.resetSizeCounter();
+    }
```

Comment:
The javadoc of `CodedInputStream#setSizeLimit` says the following:
```
If you want to read several messages from a single CodedInputStream, you could call resetSizeCounter() after each one to avoid hitting the size limit.
```
Based on that I would be inclined to reset the counter after every single message otherwise it still seems feasible to hit the same error if the `DataInput` is sufficiently large.